src/CheckoutSdk/JsonSerializer.cs

```diff
@@ -41,20 +42,33 @@ private static JsonSerializerSettings CreateSerializerSettings(Action<JsonSerial
             var settings = new JsonSerializerSettings
             {
                 NullValueHandling = NullValueHandling.Ignore,
-                ContractResolver = new DefaultContractResolver {NamingStrategy = new SnakeCaseNamingStrategy()},
+                ContractResolver = new DefaultContractResolver { NamingStrategy = new SnakeCaseNamingStrategy() },
                 Converters = new JsonConverter[]
                 {
                     new StringEnumConverter(),
                     // Instruments CS2
                     new GetInstrumentResponseTypeConverter(), new UpdateInstrumentResponseTypeConverter(),
                     // Workflows CS2
-                    new WorkflowActionTypeResponseConverter(), new WorkflowConditionTypeResponseConverter()
-                }
+                    new WorkflowActionTypeResponseConverter(), new WorkflowConditionTypeResponseConverter(),
+                    // Serialize DateTime to ISO 8601
+                    GetConverterDateTimeToIso()
+                },
             };
 
             configureSettings?.Invoke(settings);
 
             return settings;
         }
+
+        private static IsoDateTimeConverter GetConverterDateTimeToIso()
+        {
+            IsoDateTimeConverter converter = new IsoDateTimeConverter
+            {
+                DateTimeStyles = DateTimeStyles.AdjustToUniversal,
+                DateTimeFormat = "yyyy'-'MM'-'dd'T'HH':'mm':'ssK"
+            };
+
+            return converter;
```

Comment:
We should not do any DateTime conversion. The default serialization mechanism already outputs the format provided by the user. I mean, if the user provides a `DateTime.Now` the output will be `2022-01-19T17:07:18.4739344+01:00` (in my Timezone). If the user provides a `DateTime.UtcNow` the output will be `2022-01-19T10:31:12.1880839Z` containing "Z" that stands for Zulu (UTC). 

`The default format used by Json.NET is the ISO 8601 standard` [(docs)](https://www.newtonsoft.com/json/help/html/DatesInJSON.htm)

According to the documentation, dates on the body of requests should be in UTC [(example)](https://api-reference.checkout.com/#operation/retrieveEvents) but it's not up to the SDK to do that conversion. Users should provide an UTC datetime. Even if they don't, it's the responsibility of the API to validate the date format & timezone provided.

Please delete the class `DateTimeFormatConverter` I found with no usages in this project. It looks redundant.

Thoughts?

@Armando-Ibarra

Reply:
`2022-01-19T10:31:12.1880839Z` is still invalid for checkout and throws `paging_to_invalid_format`

Checkout needs this exactly format no matter what `2021-06-11T09:41:21Z`, im fine with not adding a validation , but there are only a few ways to archive this format in `natural` way, example

`DateTime uDat = new DateTime(2022, 1, 19, 13, 45, 30,DateTimeKind.Utc);`

That is not cool, and also users needs to know about this, so we should provide a better approach for users so that they can use DateTime objects.

I'm probably fine to add a converter for dates, but only for query parameters, unless we specify that a required format is necessary for query parameters dates.